### PR TITLE
implement nomad component

### PR DESCRIPTION
**Pull request by captainroy-hy**
related to #1100 
to do 
- [ ] unit test
- [ ] doc

Signed-off-by: roywang <seiwy2010@gmail.com>

### CODE REVIEW

**Comment by ryanzhang-oss** [comment]
I am not sure if this is something of high priority and we may need a bit more design on exactly what does this special case componentDef enabled.

File: docs/examples/app-with-nomad/mycloneset-adopted.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+spec:
+  components:
+  - name: mycloneset
+    properties:
+      apiVersion: apps.kruise.io/v1alpha1
+      kind: CloneSet
+      metadata:
+        name: mycloneset
+        namespace: default
+      spec:
```
Comment (resouer) [inline docs/examples/app-with-nomad/mycloneset-adopted.yaml:15]: These boilerplate lines should be removed and they should be stored in component definition.

**Comment by resouer** [comment]
@ryanzhang-oss Agree, the direction dose not fully align with the original issue #1100

@wonderflow @captainroy-hy The intention of #1100 is to enable user to use KubeVela as an independent rollout controller in his existing system. This assumption is only valid when `Rollout` becomes a highly demanded feature but user already has a legacy system and can't migrate.

**Comment by wonderflow** [comment]
closing as this is not the intended design.

**Pull request closed without merging**